src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java

```diff
@@ -297,6 +298,16 @@ private ICitizenData spawnCitizenOnPosition(
 
             colony.getEventDescriptionManager().addEventDescription(new CitizenSpawnedEvent(spawnPoint, citizenData.getName()));
         }
+
+        if (world instanceof ServerLevel serverLevel)
+        {
+            final Entity existing = serverLevel.getEntity(citizenData.getUUID());
+            if (existing != null)
+            {
+                existing.discard();
```

Comment:
I guess we should log this too

Reply:
not sure there may be some valid cases where we just re-spawn an entity